server/utils/hollaex-tools-lib/tools/broker.js

```
@@ -24,14 +32,212 @@ const validateBrokerPair = (brokerPair) => {
+				data,
+			},
+			user.settings
+		);
+	}
+	try {
+		const broker = await getModel('broker').findOne({ where: { symbol } });
+
+		if (broker.account && broker.account.hasOwnProperty('binance')) {
+			const binanceInfo = broker.account.binance;
+			const exchangeId = 'binance'
+				, exchangeClass = ccxt[exchangeId]
+				, exchange = new exchangeClass({
+					'apiKey': binanceInfo.apiKey,
+					'secret': binanceInfo.apiSecret,
+				})
+
+			const formattedSymbol = symbol.split('-').join('').toUpperCase();
+			const formattedRebalancingSymbol = broker.rebalancing_symbol && broker.rebalancing_symbol.split('-').join('').toUpperCase();
+
+			if (side === 'buy') {
+				exchange.createLimitBuyOrder(formattedRebalancingSymbol || formattedSymbol, size, price - price * 0.05)
+					.then(res => { notifyUser(res) })
+					.catch(err => { notifyUser(err) });
+			} else if (side == 'sell') {
+				exchange.createLimitSellOrder(formattedRebalancingSymbol || formattedSymbol, size, price + price * 0.05)
+					.then(res => { notifyUser(res) })
+					.catch(err => { notifyUser(err) });
+			}
+		}
+	} catch (err) {
+		loggerBroker.error(err);
+	}
+
+}
+
+const createBrokerPair = async (brokerPair) => {
 	validateBrokerPair(brokerPair);
 	return fetchBrokerPair(brokerPair.symbol)
 		.then((deal) => {
 			if (deal) {
 				throw new Error('A deal for this symbol alreadys exists')
 			}
-			return getModel("broker").create(brokerPair, { raw: true });
+			const {
+				formula,
+				exchange_name,
+				spread,
+				multiplier
+			} = brokerPair;
+
+			let adminFormula;
+
+			// If it is a custom script(users send their own formula)
+			if (formula) {
+				adminFormula = formula;
+			}
+			// If user selects a exchange
+			else if (exchange_name === 'binance') {
+				const binanceFormula = `
+					const spread = ${spread}; 
+					const multiplier = ${multiplier || 1}; 
+					module.exports = (${binanceScript.toString()})()
+				`;
+
+				adminFormula = binanceFormula;
+			}
```

@fetok12 what about after this else if? lets say its not binance and there is no formula. Front end lets say sends bitmex. the formula would now be empty I suppose which is wrong. We should throw an error instead.